frontend/src/views/CollabPageView.tsx

```diff
@@ -50,25 +54,53 @@ const customQuestion: Question = {
 	],
 };
 
-function CollabPageView() {
+const CollabPageView: React.FC = () => {
 	const [code, setCode] = useState("");
-	const [socket, setSocket] = useState<Socket | null>(null);
+   	const [socket, setSocket] = useState<Socket | null>(null);
+   	const [message, setMessage] = useState(""); // For new message input
+   	const [messages, setMessages] = useState<{ username: string; message: string }[]>([]);
+   	const [userId, setUserId] = useState<string>("");
+   	const navigate = useNavigate();
+   	const [questionData, setQuestionData] = useState<Question>(customQuestion);
+   	const { sessionId: sessionIdObj } = useParams<{ sessionId: string }>();
 	const editorRef = useRef<monaco.editor.IStandaloneCodeEditor | null>(null);
 
 	useEffect(() => {
 		// Initialize the WebSocket connection when the component mounts
 		const newSocket = io("http://localhost:5004"); // Backend WebSocket server URL
 		setSocket(newSocket);
-
+ 
 		newSocket.on("connect", () => {
 			console.log("WebSocket connected");
-			newSocket.emit("joinSession", "session123");
+			console.log("Emitting sessionJoined with sessionId:", sessionIdObj);
+			newSocket.emit("sessionJoined", sessionIdObj );
+		});
+ 
+		newSocket.on('sessionData', ({ sessionIdObj, socketId, questionData }) => {
+			sessionIdObj = sessionIdObj;
+			// Set state with the received data
+			setUserId(socketId);
+			setQuestionData(questionData);
 		});
 
+		console.log("Current user ID:", userId);
+ 
+ 
 		// Listen for code updates from the server
 		newSocket.on("codeUpdated", (data) => {
 			console.log("Code update received from server:", data);
 			setCode(data.code);
+		}); 
+
+		newSocket.on('sessionTerminated', ({ userId }) => {
```

Comment:
Could consider disconnecting the newSocket upon sessionTerminated

Reply:
have updated!